com.unity.netcode.gameobjects/Components/NetworkAnimator.cs

```diff
@@ -147,34 +172,267 @@ internal void DeregisterUpdate()
         internal NetworkAnimatorStateChangeHandler(NetworkAnimator networkAnimator)
         {
             m_NetworkAnimator = networkAnimator;
+            m_IsServer = networkAnimator.NetworkManager.IsServer;
             NetworkUpdateLoop.RegisterNetworkUpdate(this, NetworkUpdateStage.PreUpdate);
         }
     }
 
 
-
     /// <summary>
     /// NetworkAnimator enables remote synchronization of <see cref="UnityEngine.Animator"/> state for on network objects.
     /// </summary>
     [AddComponentMenu("Netcode/" + nameof(NetworkAnimator))]
     [RequireComponent(typeof(Animator))]
+    // TODO: Enable the serialization callback receiver whenever we do a minor version update.
```

Comment:
@JesseOlmer 
This was to keep this PR patch revision compliant and to keep our options open.  I left the `ISerializationCallbackReceiver` implementation here in the event we decided to do a minor version update.  If not, I can create a new branch from this PR to preserve the `ISerializationCallbackReceiver` implementation (for when we do a minor revision update) and remove the code associated with the implementation from this PR.

The `ISerializationCallbackReceiver` removes the need to figure out when we should or should not build the `TransitionStateinfo` TableData (_the name of that list is getting updated next push_).